From 3fea74401ee3f72fa444750223d713ad44557d76 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 28 Sep 2022 16:24:56 +0300
Subject: [PATCH] [SW-106171] gaudi3: remove udelay api from tpm code

As a part of tpm review, it is determined to remove udelays
in busy wait loops in order to process the completion as soon
as possible. This is because in ROM, tpm code is executed
pre pcie link up and it is required to avoid unnecessary delays.

Change-Id: I3454c1065f7e681b4abe7d03a14e14e82a533e97
---

diff --git a/arch/gaudi3/arc/tpm2.c b/arch/gaudi3/arc/tpm2.c
index 32aa8de..10c0f38 100644
--- a/arch/gaudi3/arc/tpm2.c
+++ b/arch/gaudi3/arc/tpm2.c
@@ -306,8 +306,6 @@
 
 		if ((*status & mask) == mask)
 			return 0;
-
-		udelay(MS_TO_US(TPM_TIMEOUT_MS));
 	} while (!arcboot_rom_timeout_expired(start, MS_TO_US(timeout)));
 
 	return -1;
@@ -391,7 +389,6 @@
 			return ret;
 		}
 
-		udelay(MS_TO_US(TPM_TIMEOUT_MS));
 	} while (!arcboot_rom_timeout_expired(start, MS_TO_US(TIS_SHORT_TIMEOUT_MS)));
 
 	return ret;
